### PR TITLE
Don't encode double "inf" or "-nan" or "nan"

### PR DESCRIPTION
**Pull request by stefwalter**
Previously we would encide JSON with numbers that had the words

```
  inf
  -nan
  nan
```

in them. This is invalid JSON. Instead encode these values as null.

This caused test failures, and likely failures in production as well.

### CODE REVIEW

File: src/common/cockpitjson.c [hunk 1]
```diff
@@ -720,11 +721,19 @@ dump_value (const gchar   *name,
...
-                                         json_node_get_double (node)));
+      if (fpclassify (d) == FP_NAN || fpclassify (d) == FP_INFINITE)
+        {
+          g_string_append (buffer, "null");
+        }
+      else
+        {
+          g_string_append (buffer,
+                           g_ascii_dtostr (buf, sizeof (buf),
+                                           json_node_get_double (node)));
```
Comment (mvollmer) [inline src/common/cockpitjson.c]: This could be just `d`.

Comment (stefwalter) [inline src/common/cockpitjson.c]: Indeed

Comment (stefwalter) [inline src/common/cockpitjson.c]: Done.

**Comment by stefwalter** [comment]
Fixed memory leak.

**Pull request closed without merging**